File: Daily Challanges/2026-05-May/2026-05-07.py
def get_longest_substring(s):
    n = len(s)
    window_size = len(s)-1
    memory = set()
    while window_size > 0:
        for i in range(n-window_size+1):
            window = s[i:i+window_size]
            if window in memory:
                return window
            memory.add(window)
        window_size -= 1
        memory = set()
    return s

print(get_longest_substring("abracadabra"))
print(get_longest_substring("hello world hello"))

File: Daily Challanges/2026-05-May/2026-05-07.js
function getLongestSubstring(s) {
  const n = s.length;
  let windowSize = n - 1;
  let memory = new Set();
  while (windowSize > 0) {
    for (let i = 0; i < n - windowSize + 1; i++) {
      const window = s.slice(i, i + windowSize);
      if (memory.has(window)) {
        return window;
      }
      memory.add(window);
    }
    windowSize--;
    memory = new Set();
  }
  return s;
}

console.log(getLongestSubstring("abracadabra"));
console.log(getLongestSubstring("hello world hello"));
